internal/pkg/template/templates/cicd/buildspec.yml

```diff
@@ -84,6 +84,29 @@ phases:
             jq --arg a "https://{{$bucket.BucketName}}.s3.{{$bucket.Region}}.amazonaws.com/manual/$timestamp/$workload.addons.stack.yml" '.Parameters.AddonsTemplateURL = $a' ./infrastructure/$workload-{{$envName}}.params.json > "$tmp" && mv "$tmp" ./infrastructure/$workload-{{$envName}}.params.json{{end}}{{end}}
           fi
         done;
+      # If env file exists, upload the env file to each S3 bucket and write template URL to template config files.
+      - |
+        for workload in $WORKLOADS; do
+          manifest=$(cat $CODEBUILD_SRC_DIR/copilot/$workload/manifest.yml | ruby -ryaml -rjson -e 'puts JSON.pretty_generate(YAML.load(ARGF))')
+          type=$(echo $manifest | jq -r '.type')
+          if [ "$type" = "Request-Driven Web Service" ]; then
+            echo "skipping env file uploading for Request-Driven Web Service $workload";
+            continue
+          fi
+          for env in $pl_envs; do
+            env_files=$(echo $manifest | jq --arg env "$env" '{base_env_file: .env_file, env_env_file: (.environments? | .[$env]? | .env_file?) }')
+            env_file=$(echo $env_files | jq -r 'if (.env_env_file == null) then .base_env_file else .env_env_file end')
+            ENVFILE="./$env_file"
+            if [ -f "$ENVFILE" ]; then{{range $bucket := .ArtifactBuckets}}{{range $envName := $bucket.Environments}}
```

Comment:
Does this mean we only upload the files for environment that has been registered at the time of `pipeline init`?

Reply:
Yeah that's correct.